lib/autobuild/exceptions.rb

```diff
@@ -46,6 +46,10 @@ def initialize(target = nil, phase = nil, options = Hash.new)
                 retry: false
             super(target, phase, options.merge(other_options))
         end
+
+        def to_s
+            super + "\nIf the VCS type changed, please delete your working copy first."
```

Comment:
The user has no information if someone has changes the VCS. See recommend workflow in general discussion.
Furthermore extend a generic error message with specific assumptions feels wrong.